tests/openssh.rs

```diff
@@ -718,14 +804,131 @@ async fn auth_failed() {
         String::from("ssh://openssh-tester@login.csail.mit.edu")
     };
 
-    match Session::connect(&addr, KnownHosts::Accept)
-        .await
-        .unwrap_err()
-    {
-        Error::Connect(e) => {
-            eprintln!("{:?}", e);
-            assert_eq!(e.kind(), io::ErrorKind::PermissionDenied);
+    for err in connects_err(&addr).await {
+        match err {
+            Error::Connect(e) => {
+                eprintln!("{:?}", e);
+                assert_eq!(e.kind(), io::ErrorKind::PermissionDenied);
+            }
+            e => unreachable!("{:?}", e),
         }
-        e => unreachable!("{:?}", e),
     }
 }
+
+//use core::num::NonZeroU32;
+//use core::time::Duration;
+
+//use std::fs;
+//use tokio::net::{TcpListener, TcpStream};
+//use tokio::time::sleep;
+
+// These two tests are commented out because linuxserver/openssh-server does not
+// allow port forwarding.
```

Comment:
Looking at https://github.com/linuxserver/docker-openssh-server/pull/13, it seems like the suggested way to do this is with a "[docker mod](https://mods.linuxserver.io/)". Specifically, I think you can enable it with [this one](https://github.com/linuxserver/docker-mods/tree/openssh-server-ssh-tunnel).

Reply:
Thank you for the suggestion!
I didn't know that docker images provided by linuxserver supports mods.